modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -461,7 +461,7 @@ private void removeIndentations(OMElement element, List<OMText> removables) {
                     // XML escape the result of an expression that produces a literal, if the target format
                     // of the payload is XML.
                     details.setXml(isXML(value));
-                    if (!details.isXml() && XML_TYPE.equals(getType()) && !isJson(value)) {
+                    if (!details.isXml() && XML_TYPE.equals(getType()) && !isJson(value.trim())) {
```

Comment:
Any possibility of "value" being null?

Reply:
value is null checked in https://github.com/Shakila/wso2-synapse/blob/e9e2e9c7eff4bbed9e20b0dd73c077ad185c6c9a/modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java#L460

Reply:
Yes. They have a use-case where they send spaces.

Reply:
@Nirothipan Cool.. Didn't notice that.